File: src/_03_BehavioralPattern/_03_02_Command/LightOnCommand.js
import Command from "./Command.js";
import LightOffCommand from "./LightOffCommand.js";

/**
 * @typedef { import("./Light").default } Light
 */

export default class LightOnCommand extends Command {
  /** @type { Light } */
  #light;

  /** @param { Light } light */
  constructor(light) {
    super();
    this.#light = light;
  }

  /** @override */
  execute() {
    this.#light.on();
  }

  /** @override */
  undo() {
    new LightOffCommand(this.#light).execute();
  }
}

File: src/_03_BehavioralPattern/_03_02_Command/java/after/LightOnCommand.java
package _03_BehavioralPattern._03_02_Command.java.after;

public class LightOnCommand implements Command {

  private Light light;

  public LightOnCommand(Light light) {
    this.light = light;
  }

  @Override
  public void execute() {
    this.light.on();
  }

  @Override
  public void undo() {
    new LightOffCommand(this.light).execute();
  }
  
}
